templates/repo/diff/box.tmpl

```diff
@@ -61,13 +61,11 @@
 						link: "{{$.Link}}"
 					};
 					if(window.config.pageData.diffFileInfo) {
-						// Page is already loaded - add the data to our existing data
-						window.config.pageData.diffFileInfo.files.push(...diffData.files);
-						window.config.pageData.diffFileInfo.isIncomplete = diffData.isIncomplete;
-						window.config.pageData.diffFileInfo.diffEnd = diffData.diffEnd;
-						window.config.pageData.diffFileInfo.link = diffData.link;
+						// see repo-issue.js we provide an update function
+						window.updateDiffTreeStore(diffData);
```

Comment:
I'm not sure how this can work. This script runs immediately on page load, but the `index.js` may not be loaded yet. How can it work?

Reply:
I guess a better way is to leave this data here unchanged and move the `updateDiffTreeStore` into `store.js` where it is exported and called immediately after parsing. I don't think our inline scripts should depend on functions in the main JS chunk.

Reply:
the box,tmpl is loaded multiple times. 
First time when loading the diff page -> window.config.pageData.diffFileInfo is not available (see else branch).
When clicking on Show more (for large diffs) this is loaded again -> diffFileInfo is available and set (and all js functions available). Then we just call the update function immediately.

Reply:
I see, so that's why it works. Well ideally we should find some solution that does not rely on polluting `window`. I'm sure @wxiaoguang may have some idea.

Reply:
i just removed the global func. Still testing but looks like it's working without the global